includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1482,30 +1482,48 @@ public function set_payment_method_title_for_order( $order, $payment_method_type
 	 * @return array Array of keyed metadata values.
 	 */
 	protected function get_metadata_from_order( $order, $payment_type ) {
+		switch ( get_class( $this ) ) {
+			case UPE_Split_Payment_Gateway::class:
+				$gateway_type = 'split_upe';
+				break;
+			case UPE_Payment_Gateway::class:
+				$gateway_type = 'upe';
+				break;
+			default:
+				$gateway_type = 'classic';
+		}
+
```

Comment:
Thank you for switching to `get_class( $this )` here! Looking at tests (I did not do that yesterday), I see why this `switch` might not be the best idea either, because this way we're looking for a specific class, breaking tests.

How about something simpler?

```suggestion
		if ( $this instanceof UPE_Split_Payment_Gateway ) {
			$gateway_type = 'split_upe';
		} else if ( $this instanceof UPE_Payment_Gateway ) {
			$gateway_type = 'upe';
		} else {
			$gateway_type = 'classic';
		}
```